jadx-gui/src/main/java/jadx/gui/utils/search/SearchImpl.java

```diff
@@ -0,0 +1,54 @@
+package jadx.gui.utils.search;
+
+import java.util.regex.Matcher;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class SearchImpl {
+
+	/*
+	 * Checks if searchArea matches the searched string found in searchSettings
+	 */
+	public static boolean isMatch(StringRef searchArea, final SearchSettings searchSettings) {
+		return isMatch(searchArea.toString(), searchSettings);
+	}
+
+	/*
+	 * Checks if searchArea matches the searched string found in searchSettings
+	 */
+	public static boolean isMatch(String searchArea, final SearchSettings searchSettings) {
+		return find(searchArea, searchSettings) == -1 ? false : true;
```

Comment:
This code is very likely to be misinterpreted by a developer. Why not clean and simple 

`return find(searchArea, searchSettings) != -1;`

Reply:
Thanks not sure what I was thinking